File: 858-mirror-reflection/858-mirror-reflection.py
class Solution:
    def mirrorReflection(self, p: int, q: int) -> int:
        y = math.lcm(p,q) // p 
        x = math.lcm(p,q) // q
        print(x,y)
        x, y = x%2, y%2
        print(x,y)
        if x==0 and y==1:
            return 2
        elif x == 1 and y == 1:
            return 1
        elif x == 1 and  y == 0:
            return 0
        
        
        

File: 858-mirror-reflection/858-mirror-reflection.go
func lcm(a int, b int) int{
    var temp int
    c := a * b
    for b != 0 {
        temp = a % b
        a = b
        b = temp
    }
    return c/a
}
func mirrorReflection(p int, q int) int {
    l := lcm(p,q)
    x, y := (l / q) % 2, (l / p) % 2
    if x == 0{
        return 2
    }else if y == 1{
            return 1
    }else{
        return 0
    }
}